app/controllers/merchants/items_controller.rb

```diff
@@ -1,32 +1,47 @@
 class Merchants::ItemsController < ApplicationController
+  include Dollarable
+
+  before_action :set_merchant
+  before_action :set_item, only: [:show, :edit, :update]
+
   def index
-    @merchant = Merchant.find(params[:merchant_id])
     @items = Item.where(merchant_id: @merchant.id)
   end
 
   def show
-    @merchant = Merchant.find(params[:merchant_id])
-    @item = Item.find(params[:id])
   end
 
   def edit
-    @merchant = Merchant.find(params[:merchant_id])
-    @item = Item.find(params[:id])
   end
 
   def update 
+    if params[:item][:enabled].present? && @item.update(item_params)
+      redirect_to merchant_items_path(@merchant.id)
+    else
```

Comment:
Do we want to pull out just the :enabled from this to prevent updating the item with html hidden edits on our button?

Reply:
@LawrenceWhalen I like that, thanks! Will update

Reply:
Updated ^